packages/indexer-agent/src/commands/start.ts

```diff
@@ -512,7 +512,7 @@ export default {
     // Prevent passing empty basicAuth info
     let username
     let password
-    if (providerUrl.username == '' && providerUrl.password == '') {
+    if (providerUrl.username == '' || providerUrl.password == '') {
```

Comment:
I think this should stay `&&` since there could be a case where a username string is provided and the password is an empty string.  I believe that is valid.

Reply:
Ah okay! Would it still make logical sense to not check for password at all? ie, what happens when there is empty username + non-empty password?

Reply:
See this [PR](https://github.com/graphprotocol/indexer/pull/276/files) where that was added for background. It's important to check for both because that indicates that a user did not enter anything so the headers shouldn't contain username and password.